src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java

```diff
@@ -6215,6 +6215,63 @@ public final class TokenTypes {
     public static final int PATTERN_DEF =
         JavaLanguageLexer.PATTERN_DEF;
 
+    /**
+     * A {@code when} clause. Appears as part of a switch label in a guarded pattern definition.
+     *
+     * <p>For example:</p>
+     * <pre>
+     * return switch (o) {
+     *     case Integer i when i &gt;= 0 -&gt; i;
+     *     default -&gt; 2;
+     * };
+     * </pre>
+     * <p>parses as:</p>
+     * <pre>
+     * LITERAL_RETURN -&gt; return
+     *  `--EXPR -&gt; EXPR
+     *      `--LITERAL_SWITCH -&gt; switch
+     *          |--LPAREN -&gt; (
+     *          |--EXPR -&gt; EXPR
+     *          |   `--IDENT -&gt; o
+     *          |--RPAREN -&gt; )
+     *          |--LCURLY -&gt; {
+     *          |--SWITCH_RULE -&gt; SWITCH_RULE
+     *          |   |--LITERAL_CASE -&gt; case
+     *          |   |   `--PATTERN_DEF -&gt; PATTERN_DEF
+     *          |   |       `--LITERAL_WHEN -&gt; when
+     *          |   |           |--PATTERN_VARIABLE_DEF -&gt; PATTERN_VARIABLE_DEF
+     *          |   |           |   |--MODIFIERS -&gt; MODIFIERS
+     *          |   |           |   |--TYPE -&gt; TYPE
+     *          |   |           |   |   `--IDENT -&gt; Integer
+     *          |   |           |   `--IDENT -&gt; i
+     *          |   |           `--GE -&gt; &gt;=
+     *          |   |               |--IDENT -&gt; i
+     *          |   |               `--NUM_INT -&gt; 0
+     *          |   |--LAMBDA -&gt; -&gt;
+     *          |   |--EXPR -&gt; EXPR
+     *          |   |   `--IDENT -&gt; i
+     *          |   `--SEMI -&gt; ;
+     *          |--SWITCH_RULE -&gt; SWITCH_RULE
+     *          |   |--LITERAL_DEFAULT -&gt; default
+     *          |   |--LAMBDA -&gt; -&gt;
+     *          |   |--EXPR -&gt; EXPR
+     *          |   |   `--NUM_INT -&gt; 2
+     *          |   `--SEMI -&gt; ;
+     *          `--RCURLY -&gt; }
+     * </pre>
+     *
+     * @see <a href="https://docs.oracle.com/javase/specs/jls/se17/html/jls-14.html#jls-14.30">
+     * Java Language Specification, &sect;14.30</a>
+     * @see #LITERAL_SWITCH
+     * @see #PATTERN_VARIABLE_DEF
+     * @see #LITERAL_INSTANCEOF
+     * @see #SWITCH_RULE
+     *
+     * @since 10.6.1
```

Comment:
updated, unless this PR makes 10.7? I am not sure about Checkstyle emotional versioning :)

Reply:
@romani ping. It seems like we need a process when versioning number is changed from the original plan.

Reply:
We have only first digit as emotional. All others are strictly following semiver.

Reply:
Ok, then this update should be 10.7 (from https://semver.org/):

> Given a version number MAJOR.MINOR.PATCH, increment the:

>MAJOR version when you make incompatible API changes
MINOR version when you add functionality in a backwards compatible manner
PATCH version when you make backwards compatible bug fixes
Additional labels for pre-release and build metadata are available as extensions to the MAJOR.MINOR.PATCH format.

Reply:
We need to be more attentive, as after merge of any fix for breaking compatibility, it will be 10.7.0

Reply:
Updated to 10.7.0 since we have some bug fixes